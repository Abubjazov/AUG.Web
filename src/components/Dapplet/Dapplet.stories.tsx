import type { Meta, StoryObj } from '@storybook/react'
import {
  mockCommunityTags,
  mockDapplets,
  mockUserTags,
} from 'mockData/mockData'
import MockedProvider from 'mockData/mockedReduxProvider'

import Dapplet, { DappletProps } from './Dapplet'

import '/src/index.css'

const meta: Meta<DappletProps> = {
  component: Dapplet,
  title: 'COMPONENTS/Dapplet',
  parameters: {
    docs: {
      description: {
        component: 'Interface component - Dapplet',
      },
    },
  },
  tags: ['autodocs'],
  argTypes: {
    userStyles: {
      description: 'Passing additional custom styles for root element',
    },
    dapplet: { description: 'Dapplet data' },
    dappletUserTags: { description: 'User tags for this dapplet' },
    dappletCommunityTags: { description: 'Community tags for this dapplet' },
    dragOver: {
      description: 'Optional drag-over handler',
    },
  },
}

export default meta

type Story = StoryObj<DappletProps>

export const Default: Story = {
  args: {
    dapplet: mockDapplets[0],
    dappletUserTags: mockUserTags.slice(0, 3),
    dappletCommunityTags: mockCommunityTags.slice(0, 2),
  },
  decorators: [
    Story => (
      <MockedProvider>
        <Story />
      </MockedProvider>
    ),
  ],
}
