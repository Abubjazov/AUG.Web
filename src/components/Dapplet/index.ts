import Dapplet from './Dapplet'

export default Dapplet
