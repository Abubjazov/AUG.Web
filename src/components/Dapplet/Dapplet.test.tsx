import { PropsWithChildren } from 'react'

import { configureStore } from '@reduxjs/toolkit'
import { screen, fireEvent, render } from '@testing-library/react'
import {
  mockDapplets,
  mockUserDapplets,
  mockUserLists,
  mockUserTags,
} from 'mockData/mockData'
import { mockedReduxProvider as MProvider } from 'mockData/mockedReduxProvider'
import { Provider } from 'react-redux'
import userDataSliceReducer, {
  EDappletOperation,
} from 'store/slices/userDataSlice'

import Dapplet from './Dapplet'

describe('Dapplet', () => {
  test('should render Dapplet default', () => {
    const { asFragment } = render(
      <MProvider>
        <Dapplet
          dapplet={mockDapplets[3]}
          dappletUserTags={''}
          dappletCommunityTags={''}
        />
      </MProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render Dapplet then windowInnerWidth <= 880', () => {
    window.innerWidth = 880

    const { asFragment } = render(
      <MProvider>
        <Dapplet
          dapplet={mockDapplets[3]}
          dappletUserTags={''}
          dappletCommunityTags={''}
        />
      </MProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render Dapplet default after clicking Burger button', () => {
    window.innerWidth = 1601

    const { asFragment } = render(
      <MProvider>
        <Dapplet
          dapplet={mockDapplets[3]}
          dappletUserTags={''}
          dappletCommunityTags={''}
        />
      </MProvider>,
    )

    expect(asFragment()).toMatchSnapshot()

    fireEvent.click(screen.getByTestId('dapplet-burger-button'))

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render Dapplet when the "dappletOperationGoing"', () => {
    const mokedStore = configureStore({
      reducer: {
        userData: userDataSliceReducer,
      },
      preloadedState: {
        userData: {
          userDapplets: mockUserDapplets,
          userTags: mockUserTags,
          userLists: mockUserLists,
          isAddingUserTag: false,
          isAddingUserList: false,
          isLoadingUserData: false,
          tagOperationGoing: [],
          dappletOperationGoing: [
            {
              dappletId: 'ECNk2nNngwGXouvMpjWt',
              operation: EDappletOperation.INSTALL,
            },
          ],
          listOperationGoing: [],
        },
      },
    })

    const NewProvider = ({ children }: PropsWithChildren<object>) => {
      return <Provider store={mokedStore}>{children}</Provider>
    }

    const { asFragment } = render(
      <NewProvider>
        <Dapplet
          dapplet={mockDapplets[0]}
          dappletUserTags={''}
          dappletCommunityTags={''}
        />
      </NewProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render Dapplet after clicking on dapplet then windowInnerWidth <= 880', () => {
    window.innerWidth = 880

    const { asFragment } = render(
      <MProvider>
        <Dapplet
          dapplet={mockDapplets[3]}
          dappletUserTags={''}
          dappletCommunityTags={''}
        />
      </MProvider>,
    )

    expect(asFragment()).toMatchSnapshot()

    fireEvent.click(screen.getByTestId('dapplet'))

    expect(asFragment()).toMatchSnapshot()
  })

  test('should render Dapplet when the "dappletOperationGoing"', () => {
    const mokedStore = configureStore({
      reducer: {
        userData: userDataSliceReducer,
      },
      preloadedState: {
        userData: {
          userDapplets: mockUserDapplets,
          userTags: mockUserTags,
          userLists: mockUserLists,
          isAddingUserTag: false,
          isAddingUserList: false,
          isLoadingUserData: false,
          tagOperationGoing: [],
          dappletOperationGoing: [
            {
              dappletId: 'ECNk2nNngwGXouvMpjWt',
              operation: EDappletOperation.UNINSTALL,
            },
          ],
          listOperationGoing: [],
        },
      },
    })

    const NewProvider = ({ children }: PropsWithChildren<object>) => {
      return <Provider store={mokedStore}>{children}</Provider>
    }

    const { asFragment } = render(
      <NewProvider>
        <Dapplet
          dapplet={mockDapplets[0]}
          dappletUserTags={''}
          dappletCommunityTags={''}
        />
      </NewProvider>,
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
