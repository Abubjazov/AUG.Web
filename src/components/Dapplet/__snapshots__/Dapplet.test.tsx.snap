// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Dapplet > should render Dapplet after clicking on dapplet then windowInnerWidth <= 880 1`] = `
<DocumentFragment>
  <div
    class="_root_fe1661 "
    data-testid="dapplet"
  >
    <div
      class="_main-part_fe1661"
    >
      <div
        class="_main-part-header_fe1661"
      >
        <div
          class="_main-part-logo-wrapper_fe1661"
        >
          <img
            alt="Company\`s logo"
            class="_logo_fe1661"
            src="/public/images/notAvailable.svg"
          />
          <div
            class="_dapplet-name-wrapper_fe1661"
          >
            <span
              class="_dapplet-name_fe1661"
            >
              XRP Ledger
            </span>
            <span
              class="_dapplet-owner_fe1661"
            >
              XRP Ledger
            </span>
          </div>
        </div>
        <div
          class="_dapplet-installbuttons-wrapper_fe1661"
        >
          <button
            aria-label="Install dapplet T6hUx4HWCKtnIEfwQxYp"
            class="_root_cc4fb1 _mobile_cc4fb1 _installed_cc4fb1 "
            data-testid="install-button"
            type="button"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_8162_399)"
              >
                <path
                  d="M12.8613 6.66681V7.23031C12.8606 8.55112 12.4329 9.8363 11.642 10.8942C10.8512 11.9521 9.73959 12.726 8.47298 13.1004C7.20638 13.4749 5.85265 13.43 4.61369 12.9722C3.37473 12.5145 2.31693 11.6685 1.59804 10.5605C0.879154 9.45248 0.5377 8.14175 0.624605 6.8238C0.71151 5.50586 1.22212 4.25131 2.08027 3.24727C2.93843 2.24323 4.09816 1.54348 5.38649 1.2524C6.67483 0.96132 8.02274 1.09449 9.2292 1.63206"
                  stroke="white"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <path
                  d="M12.8613 1.97681L6.80364 8.97681L4.98633 6.8789"
                  stroke="white"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_8162_399"
                >
                  <rect
                    fill="white"
                    height="14"
                    width="14"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
        </div>
      </div>
      <span
        class="_dapplet-descriptor_fe1661"
      >
        Unlike Bitcoin or Ethereum, the XRPL uses a unique Federated Consensus mechanism as its method of validating transactions.
      </span>
      <div
        class="_root_dea2c3 "
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dapplet > should render Dapplet after clicking on dapplet then windowInnerWidth <= 880 2`] = `
<DocumentFragment>
  <div
    class="_root_fe1661 "
    data-testid="dapplet"
  >
    <div
      class="_main-part_fe1661"
    >
      <div
        class="_main-part-header_fe1661"
      >
        <div
          class="_main-part-logo-wrapper_fe1661"
        >
          <img
            alt="Company\`s logo"
            class="_logo_fe1661"
            src="/public/images/notAvailable.svg"
          />
          <div
            class="_dapplet-name-wrapper_fe1661"
          >
            <span
              class="_dapplet-name_fe1661"
            >
              XRP Ledger
            </span>
            <span
              class="_dapplet-owner_fe1661"
            >
              XRP Ledger
            </span>
          </div>
        </div>
        <div
          class="_dapplet-installbuttons-wrapper_fe1661"
        >
          <button
            aria-label="Install dapplet T6hUx4HWCKtnIEfwQxYp"
            class="_root_cc4fb1 _mobile_cc4fb1 _installed_cc4fb1 "
            data-testid="install-button"
            type="button"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_8162_399)"
              >
                <path
                  d="M12.8613 6.66681V7.23031C12.8606 8.55112 12.4329 9.8363 11.642 10.8942C10.8512 11.9521 9.73959 12.726 8.47298 13.1004C7.20638 13.4749 5.85265 13.43 4.61369 12.9722C3.37473 12.5145 2.31693 11.6685 1.59804 10.5605C0.879154 9.45248 0.5377 8.14175 0.624605 6.8238C0.71151 5.50586 1.22212 4.25131 2.08027 3.24727C2.93843 2.24323 4.09816 1.54348 5.38649 1.2524C6.67483 0.96132 8.02274 1.09449 9.2292 1.63206"
                  stroke="white"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <path
                  d="M12.8613 1.97681L6.80364 8.97681L4.98633 6.8789"
                  stroke="white"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_8162_399"
                >
                  <rect
                    fill="white"
                    height="14"
                    width="14"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
          <button
            aria-label="Install dapplet T6hUx4HWCKtnIEfwQxYp"
            class="_root_cc4fb1 _mobile_cc4fb1 _uninstall_cc4fb1 "
            data-testid="install-button"
            type="button"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.00008 12.5681C10.1782 12.5681 12.7546 10.0752 12.7546 6.99994C12.7546 3.92472 10.1782 1.43176 7.00008 1.43176C3.82197 1.43176 1.24561 3.92472 1.24561 6.99994C1.24561 10.0752 3.82197 12.5681 7.00008 12.5681Z"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M8.72612 5.32959L5.27344 8.6705"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M5.27344 5.32959L8.72612 8.6705"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
        </div>
      </div>
      <span
        class="_dapplet-descriptor_fe1661"
      >
        Unlike Bitcoin or Ethereum, the XRPL uses a unique Federated Consensus mechanism as its method of validating transactions.
      </span>
      <div
        class="_root_dea2c3 "
      >
        <button
          class="_button_dea2c3"
          data-testid="add-tag-button"
          type="button"
        >
          +
        </button>
      </div>
    </div>
    <div
      class="_additional-part_fe1661"
    >
      <div
        class="_root_94add5 _main-descriptor_fe1661"
      >
        <span
          class="_title_94add5"
        >
          About
        </span>
        <span
          class="_text_94add5"
        >
          Launched in 2021, the XRP Ledger (XRPL) is an open-source, permissionless and decentralized technology. Benefits of the XRP Ledger include its low-cost ($0.0002 to transact), speed (settling transactions in 3-5 seconds), scalability (1,500 transactions per second) and inherently green attributes (carbon-neutral and energy-efficient). The XRP Ledger also features the first decentralized exchange (DEX) and custom tokenization capabilities built into the protocol. Since 2012, the XRP Ledger has been operating reliably, having closed 70 million ledgers.
        </span>
      </div>
      <div
        class="_additional-fields_fe1661"
      >
        <div
          class="_root_951cff"
        >
          <span
            class="_title_951cff"
          >
            Market cap
          </span>
          <div
            class="_wrapper_951cff _additional-descriptor_fe1661 _green_951cff"
          >
            <svg
              fill="#14C785"
              height="15px"
              viewBox="0 0 24 24"
              width="15px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.0566 16H5.94336C5.10459 16 4.68455 14.9782 5.27763 14.3806L11.3343 8.27783C11.7019 7.90739 12.2981 7.90739 12.6657 8.27783L18.7223 14.3806C19.3155 14.9782 18.8954 16 18.0566 16Z"
              />
            </svg>
            5.66 %
          </div>
        </div>
        <div
          class="_root_951cff"
        >
          <span
            class="_title_951cff"
          >
            Volume (24h)
          </span>
          <div
            class="_wrapper_951cff _additional-descriptor_fe1661 _green_951cff"
          >
            <svg
              fill="#14C785"
              height="15px"
              viewBox="0 0 24 24"
              width="15px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.0566 16H5.94336C5.10459 16 4.68455 14.9782 5.27763 14.3806L11.3343 8.27783C11.7019 7.90739 12.2981 7.90739 12.6657 8.27783L18.7223 14.3806C19.3155 14.9782 18.8954 16 18.0566 16Z"
              />
            </svg>
            5.66 %
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dapplet > should render Dapplet default 1`] = `
<DocumentFragment>
  <div
    class="_root_fe1661 "
    data-testid="dapplet"
  >
    <div
      class="_main-part_fe1661"
    >
      <div
        class="_burger_fe1661"
        data-testid="dapplet-burger-button"
      >
        <svg
          fill="none"
          height="10"
          viewBox="0 0 18 10"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="0.5"
          />
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="4.5"
          />
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="8.5"
          />
        </svg>
      </div>
      <img
        alt="Company\`s logo"
        class="_logo_fe1661"
        src="/public/images/notAvailable.svg"
      />
      <div
        class="_dapplet-name-wrapper_fe1661"
      >
        <span
          class="_dapplet-name_fe1661"
        >
          XRP Ledger XRP
        </span>
        <div
          class="_root_951cff"
        >
          <div
            class="_wrapper_951cff _dapplet-market-cap_fe1661 _green_951cff"
          >
            <svg
              fill="#14C785"
              height="15px"
              viewBox="0 0 24 24"
              width="15px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.0566 16H5.94336C5.10459 16 4.68455 14.9782 5.27763 14.3806L11.3343 8.27783C11.7019 7.90739 12.2981 7.90739 12.6657 8.27783L18.7223 14.3806C19.3155 14.9782 18.8954 16 18.0566 16Z"
              />
            </svg>
            5.66 %
          </div>
        </div>
      </div>
      <span
        class="_dapplet-descriptor_fe1661"
      >
        Unlike Bitcoin or Ethereum, the XRPL uses a unique Federated Consensus mechanism as its method of validating transactions.
      </span>
      <span
        class="_dapplet-owner_fe1661"
      >
        XRP Ledger
      </span>
      <div
        class="_root_dea2c3 "
      />
      <button
        aria-label="Install dapplet T6hUx4HWCKtnIEfwQxYp"
        class="_root_cc4fb1 _desktop_cc4fb1 _installed_cc4fb1 "
        data-testid="install-button"
        type="button"
      >
        installed
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dapplet > should render Dapplet default after clicking Burger button 1`] = `
<DocumentFragment>
  <div
    class="_root_fe1661 "
    data-testid="dapplet"
  >
    <div
      class="_main-part_fe1661"
    >
      <div
        class="_burger_fe1661"
        data-testid="dapplet-burger-button"
      >
        <svg
          fill="none"
          height="10"
          viewBox="0 0 18 10"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="0.5"
          />
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="4.5"
          />
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="8.5"
          />
        </svg>
      </div>
      <img
        alt="Company\`s logo"
        class="_logo_fe1661"
        src="/public/images/notAvailable.svg"
      />
      <div
        class="_dapplet-name-wrapper_fe1661"
      >
        <span
          class="_dapplet-name_fe1661"
        >
          XRP Ledger XRP
        </span>
        <div
          class="_root_951cff"
        >
          <div
            class="_wrapper_951cff _dapplet-market-cap_fe1661 _green_951cff"
          >
            <svg
              fill="#14C785"
              height="15px"
              viewBox="0 0 24 24"
              width="15px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.0566 16H5.94336C5.10459 16 4.68455 14.9782 5.27763 14.3806L11.3343 8.27783C11.7019 7.90739 12.2981 7.90739 12.6657 8.27783L18.7223 14.3806C19.3155 14.9782 18.8954 16 18.0566 16Z"
              />
            </svg>
            5.66 %
          </div>
        </div>
      </div>
      <span
        class="_dapplet-descriptor_fe1661"
      >
        Unlike Bitcoin or Ethereum, the XRPL uses a unique Federated Consensus mechanism as its method of validating transactions.
      </span>
      <span
        class="_dapplet-owner_fe1661"
      >
        XRP Ledger
      </span>
      <div
        class="_root_dea2c3 "
      />
      <button
        aria-label="Install dapplet T6hUx4HWCKtnIEfwQxYp"
        class="_root_cc4fb1 _desktop_cc4fb1 _installed_cc4fb1 "
        data-testid="install-button"
        type="button"
      >
        installed
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dapplet > should render Dapplet default after clicking Burger button 2`] = `
<DocumentFragment>
  <div
    class="_root_fe1661 "
    data-testid="dapplet"
  >
    <div
      class="_main-part_fe1661"
    >
      <div
        class="_burger_fe1661"
        data-testid="dapplet-burger-button"
      >
        <svg
          fill="none"
          height="10"
          viewBox="0 0 18 10"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="0.5"
          />
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="4.5"
          />
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="8.5"
          />
        </svg>
      </div>
      <img
        alt="Company\`s logo"
        class="_logo_fe1661"
        src="/public/images/notAvailable.svg"
      />
      <div
        class="_dapplet-name-wrapper_fe1661"
      >
        <span
          class="_dapplet-name_fe1661"
        >
          XRP Ledger XRP
        </span>
        <div
          class="_root_951cff"
        >
          <div
            class="_wrapper_951cff _dapplet-market-cap_fe1661 _green_951cff"
          >
            <svg
              fill="#14C785"
              height="15px"
              viewBox="0 0 24 24"
              width="15px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.0566 16H5.94336C5.10459 16 4.68455 14.9782 5.27763 14.3806L11.3343 8.27783C11.7019 7.90739 12.2981 7.90739 12.6657 8.27783L18.7223 14.3806C19.3155 14.9782 18.8954 16 18.0566 16Z"
              />
            </svg>
            5.66 %
          </div>
        </div>
      </div>
      <span
        class="_dapplet-descriptor_fe1661"
      >
        Unlike Bitcoin or Ethereum, the XRPL uses a unique Federated Consensus mechanism as its method of validating transactions.
      </span>
      <span
        class="_dapplet-owner_fe1661"
      >
        XRP Ledger
      </span>
      <div
        class="_root_dea2c3 "
      />
      <button
        aria-label="Install dapplet T6hUx4HWCKtnIEfwQxYp"
        class="_root_cc4fb1 _desktop_cc4fb1 _installed_cc4fb1 "
        data-testid="install-button"
        type="button"
      >
        installed
      </button>
    </div>
    <div
      class="_additional-part_fe1661"
    >
      <div
        class="_root_94add5 _main-descriptor_fe1661"
      >
        <span
          class="_title_94add5"
        >
          About
        </span>
        <span
          class="_text_94add5"
        >
          Launched in 2021, the XRP Ledger (XRPL) is an open-source, permissionless and decentralized technology. Benefits of the XRP Ledger include its low-cost ($0.0002 to transact), speed (settling transactions in 3-5 seconds), scalability (1,500 transactions per second) and inherently green attributes (carbon-neutral and energy-efficient). The XRP Ledger also features the first decentralized exchange (DEX) and custom tokenization capabilities built into the protocol. Since 2012, the XRP Ledger has been operating reliably, having closed 70 million ledgers.
        </span>
      </div>
      <div
        class="_additional-fields_fe1661"
      >
        <div
          class="_root_951cff"
        >
          <span
            class="_title_951cff"
          >
            Market cap
          </span>
          <div
            class="_wrapper_951cff _additional-descriptor_fe1661 _green_951cff"
          >
            <svg
              fill="#14C785"
              height="15px"
              viewBox="0 0 24 24"
              width="15px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.0566 16H5.94336C5.10459 16 4.68455 14.9782 5.27763 14.3806L11.3343 8.27783C11.7019 7.90739 12.2981 7.90739 12.6657 8.27783L18.7223 14.3806C19.3155 14.9782 18.8954 16 18.0566 16Z"
              />
            </svg>
            5.66 %
          </div>
        </div>
        <div
          class="_root_951cff"
        >
          <span
            class="_title_951cff"
          >
            Volume (24h)
          </span>
          <div
            class="_wrapper_951cff _additional-descriptor_fe1661 _green_951cff"
          >
            <svg
              fill="#14C785"
              height="15px"
              viewBox="0 0 24 24"
              width="15px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.0566 16H5.94336C5.10459 16 4.68455 14.9782 5.27763 14.3806L11.3343 8.27783C11.7019 7.90739 12.2981 7.90739 12.6657 8.27783L18.7223 14.3806C19.3155 14.9782 18.8954 16 18.0566 16Z"
              />
            </svg>
            132.26 %
          </div>
        </div>
        <div
          class="_root_94add5 _additional-descriptor_fe1661"
        >
          <span
            class="_title_94add5"
          >
            Volume/Market cap (24h)
          </span>
          <span
            class="_text_94add5"
          >
            5.37 %
          </span>
        </div>
        <div
          class="_root_94add5 _additional-descriptor_fe1661"
        >
          <span
            class="_title_94add5"
          >
            Circulating supply
          </span>
          <span
            class="_text_94add5"
          >
            53,615,837,759 XRP
          </span>
        </div>
        <div
          class="_root_94add5 _additional-descriptor_fe1661"
        >
          <span
            class="_title_94add5"
          >
            Total supply
          </span>
          <span
            class="_text_94add5"
          >
            99,988,316,618 XRP
          </span>
        </div>
        <div
          class="_root_94add5 _additional-descriptor_fe1661"
        >
          <span
            class="_title_94add5"
          >
            Max. supply
          </span>
          <span
            class="_text_94add5"
          >
            100,000,000,000 XRP
          </span>
        </div>
        <div
          class="_root_94add5 _additional-descriptor_fe1661"
        >
          <span
            class="_title_94add5"
          >
            Fully diluted market cap
          </span>
          <span
            class="_text_94add5"
          >
            65,183,420,834 $
          </span>
        </div>
        <div
          class="_root_94add5 _additional-descriptor_fe1661"
        >
          <span
            class="_title_94add5"
          >
            Launched at
          </span>
          <span
            class="_text_94add5"
          >
            5/14/2021
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dapplet > should render Dapplet then windowInnerWidth <= 880 1`] = `
<DocumentFragment>
  <div
    class="_root_fe1661 "
    data-testid="dapplet"
  >
    <div
      class="_main-part_fe1661"
    >
      <div
        class="_main-part-header_fe1661"
      >
        <div
          class="_main-part-logo-wrapper_fe1661"
        >
          <img
            alt="Company\`s logo"
            class="_logo_fe1661"
            src="/public/images/notAvailable.svg"
          />
          <div
            class="_dapplet-name-wrapper_fe1661"
          >
            <span
              class="_dapplet-name_fe1661"
            >
              XRP Ledger
            </span>
            <span
              class="_dapplet-owner_fe1661"
            >
              XRP Ledger
            </span>
          </div>
        </div>
        <div
          class="_dapplet-installbuttons-wrapper_fe1661"
        >
          <button
            aria-label="Install dapplet T6hUx4HWCKtnIEfwQxYp"
            class="_root_cc4fb1 _mobile_cc4fb1 _installed_cc4fb1 "
            data-testid="install-button"
            type="button"
          >
            <svg
              fill="none"
              height="14"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_8162_399)"
              >
                <path
                  d="M12.8613 6.66681V7.23031C12.8606 8.55112 12.4329 9.8363 11.642 10.8942C10.8512 11.9521 9.73959 12.726 8.47298 13.1004C7.20638 13.4749 5.85265 13.43 4.61369 12.9722C3.37473 12.5145 2.31693 11.6685 1.59804 10.5605C0.879154 9.45248 0.5377 8.14175 0.624605 6.8238C0.71151 5.50586 1.22212 4.25131 2.08027 3.24727C2.93843 2.24323 4.09816 1.54348 5.38649 1.2524C6.67483 0.96132 8.02274 1.09449 9.2292 1.63206"
                  stroke="white"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
                <path
                  d="M12.8613 1.97681L6.80364 8.97681L4.98633 6.8789"
                  stroke="white"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_8162_399"
                >
                  <rect
                    fill="white"
                    height="14"
                    width="14"
                  />
                </clippath>
              </defs>
            </svg>
          </button>
        </div>
      </div>
      <span
        class="_dapplet-descriptor_fe1661"
      >
        Unlike Bitcoin or Ethereum, the XRPL uses a unique Federated Consensus mechanism as its method of validating transactions.
      </span>
      <div
        class="_root_dea2c3 "
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dapplet > should render Dapplet when the "dappletOperationGoing" 1`] = `
<DocumentFragment>
  <div
    class="_root_fe1661 "
    data-testid="dapplet"
  >
    <div
      class="_main-part_fe1661"
    >
      <div
        class="_burger_fe1661"
        data-testid="dapplet-burger-button"
      >
        <svg
          fill="none"
          height="10"
          viewBox="0 0 18 10"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="0.5"
          />
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="4.5"
          />
          <rect
            fill="#BBBBBB"
            height="1"
            width="18"
            y="8.5"
          />
        </svg>
      </div>
      <img
        alt="Company\`s logo"
        class="_logo_fe1661"
        src="/public/images/notAvailable.svg"
      />
      <div
        class="_dapplet-name-wrapper_fe1661"
      >
        <span
          class="_dapplet-name_fe1661"
        >
          Bitcoin Contracts BTC
        </span>
        <div
          class="_root_951cff"
        >
          <div
            class="_wrapper_951cff _dapplet-market-cap_fe1661 _red_951cff"
          >
            <svg
              class="sc-aef7b723-0 eummbo"
              fill="#c50140"
              height="15px"
              viewBox="0 0 24 24"
              width="15px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18.0566 8H5.94336C5.10459 8 4.68455 9.02183 5.27763 9.61943L11.3343 15.7222C11.7019 16.0926 12.2981 16.0926 12.6657 15.7222L18.7223 9.61943C19.3155 9.02183 18.8954 8 18.0566 8Z"
              />
            </svg>
            -1.01 %
          </div>
        </div>
      </div>
      <span
        class="_dapplet-descriptor_fe1661"
      >
        Bitcoin’s original inventor is known under a pseudonym, Satoshi Nakamoto. 
      </span>
      <span
        class="_dapplet-owner_fe1661"
      >
        Satoshi Nakamoto
      </span>
      <div
        class="_root_dea2c3 "
      />
      <button
        aria-label="Install dapplet ECNk2nNngwGXouvMpjWt"
        class="_root_cc4fb1 _desktop_cc4fb1 _installed_cc4fb1 "
        data-testid="install-button"
        type="button"
      >
        <svg
          class="small-spinner"
          height="20"
          preserveAspectRatio="xMidYMid"
          viewBox="0 0 100 100"
          width="20"
        >
          <path
            d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
            fill="none"
            stroke="#fff"
            stroke-dasharray="42.76482137044271 42.76482137044271"
            stroke-linecap="round"
            stroke-width="8"
          >
            <animate
              attributeName="stroke-dashoffset"
              dur="1s"
              keyTimes="0;1"
              repeatCount="indefinite"
              values="0;256.58892822265625"
            />
          </path>
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Dapplet > should render Dapplet when the "dappletOperationGoing" 2`] = `
<DocumentFragment>
  <div
    class="_root_fe1661 "
    data-testid="dapplet"
  >
    <div
      class="_main-part_fe1661"
    >
      <div
        class="_main-part-header_fe1661"
      >
        <div
          class="_main-part-logo-wrapper_fe1661"
        >
          <img
            alt="Company\`s logo"
            class="_logo_fe1661"
            src="/public/images/notAvailable.svg"
          />
          <div
            class="_dapplet-name-wrapper_fe1661"
          >
            <span
              class="_dapplet-name_fe1661"
            >
              Bitcoin Contracts
            </span>
            <span
              class="_dapplet-owner_fe1661"
            >
              Satoshi Nakamoto
            </span>
          </div>
        </div>
        <div
          class="_dapplet-installbuttons-wrapper_fe1661"
        >
          <button
            aria-label="Install dapplet ECNk2nNngwGXouvMpjWt"
            class="_root_cc4fb1 _mobile_cc4fb1 _installed_cc4fb1 "
            data-testid="install-button"
            type="button"
          >
            <svg
              class="small-spinner"
              height="20"
              preserveAspectRatio="xMidYMid"
              viewBox="0 0 100 100"
              width="20"
            >
              <path
                d="M24.3 30C11.4 30 5 43.3 5 50s6.4 20 19.3 20c19.3 0 32.1-40 51.4-40 C88.6 30 95 43.3 95 50s-6.4 20-19.3 20C56.4 70 43.6 30 24.3 30z"
                fill="none"
                stroke="#fff"
                stroke-dasharray="42.76482137044271 42.76482137044271"
                stroke-linecap="round"
                stroke-width="8"
              >
                <animate
                  attributeName="stroke-dashoffset"
                  dur="1s"
                  keyTimes="0;1"
                  repeatCount="indefinite"
                  values="0;256.58892822265625"
                />
              </path>
            </svg>
          </button>
        </div>
      </div>
      <span
        class="_dapplet-descriptor_fe1661"
      >
        Bitcoin’s original inventor is known under a pseudonym, Satoshi Nakamoto. 
      </span>
      <div
        class="_root_dea2c3 "
      />
    </div>
  </div>
</DocumentFragment>
`;
